packages/cli/src/transform.js

```diff
@@ -94,16 +100,56 @@ export async function compileFile(
 ): Promise<?string> {
   const inputFilePath = path.join(config.input, filePath);
   const outputFilePath = path.join(config.output, filePath);
+  const cachePath = getDefaultCachePath();
+
+  const inputHash = computeHash(inputFilePath);
```

Comment:
Yup. Making things async will be an improvement as you can compute the hash *and* read the cache *and* check if the output file exists, all in parallel.